Services/Certificate/classes/Placeholder/Description/class.ilCoursePlaceholderDescription.php

```diff
@@ -1,43 +1,22 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCoursePlaceholderDescription implements ilCertificatePlaceholderDescription
 {
-    /**
-     * @var ilDefaultPlaceholderDescription
-     */
-    private $defaultPlaceHolderDescriptionObject;
-
-    /**
-     * @var ilObjectCustomUserFieldsPlaceholderDescription
-     */
-    private $customUserFieldsPlaceholderDescriptionObject;
-    /**
-     * @var ilLanguage|null
-     */
-    private $language;
-
-    /**
-     * @var array
-     */
-    private $placeholder;
+    private ?ilDefaultPlaceholderDescription $defaultPlaceHolderDescriptionObject;
+    private ?ilObjectCustomUserFieldsPlaceholderDescription $customUserFieldsPlaceholderDescriptionObject;
+    private ?ilLanguage $language;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.